lib/modules/platform/gitlab/index.ts

```diff
@@ -855,10 +855,39 @@ export async function updatePr({
     body.remove_labels = removeLabels;
   }
 
-  await gitlabApi.putJson(
-    `projects/${config.repository}/merge_requests/${iid}`,
-    { body },
-  );
+  const updatedPrInfo = (
+    await gitlabApi.putJson<{
+      iid: number;
+      source_branch: string;
+      title: string;
+      state: string;
+      created_at: string;
+    }>(`projects/${config.repository}/merge_requests/${iid}`, { body })
+  ).body;
+
+  const updatedPr: Pr = massagePr({
+    number: updatedPrInfo.iid,
+    state: updatedPrInfo.state === 'opened' ? 'open' : updatedPrInfo.state,
+    title: updatedPrInfo.title,
+    createdAt: updatedPrInfo.created_at,
+    sourceBranch: updatedPrInfo.source_branch,
+  });
```

Comment:
Similar code to this seems duplicated throughout this gitlab module. Can any of it be consolidated?

Reply:
https://github.com/renovatebot/renovate/pull/32428